### PR TITLE
feat: Japanese translation for docs

### PR DESCRIPTION
**Pull request by yoshi2no**
## Changes

ref: #1815 

- fix: correct link to LICENSE (en, zh)
- fix: correct link to openapi-fetch, `Enable noUncheckedIndexedAccess in TSConfig`   (en, zh)
- add vitepress `ja` config
- translate openapi-typescript
- translate openapi-fetch
- translate openapi-react-query

I believe there shouldn’t be any content missing from the latest documents since I’m referring to the most up-to-date version.

## How to Review

- language code is correct
  - [ja, 日本語](https://localizely.com/language-code/ja/)
- It should redirect to version 6.x of the English documentation when using version 6.x with the Japanese language specified
- Ensure the "search docs" is working
  - ~~Is there anything I need to do to enable this for the Japanese docs?~~

## Checklist
N/A

### CODE REVIEW

**Comment by drwpow** [comment]
Thank you so much for opening! Re: search, that’s all automatic! Algolia scans the published site on its own, so there’s nothing to do on our end. Yes there is a slight delay before Japenese docs would appear in search yes but not long (~1 day)

Is it OK if I push an empty commit to
this PR? That’ll create a preview link on Cloudflare Pages to test out

**Comment by yoshi2no** [comment]
@drwpow 

ok! yes please!

File: docs/introduction.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ description: Quickstart
 
 openapi-typescript turns [OpenAPI 3.0 & 3.1](https://spec.openapis.org/oas/latest.html) schemas into TypeScript quickly using Node.js. No Java/node-gyp/running OpenAPI servers necessary.
 
-The code is [MIT-licensed](https://github.com/openapi-ts/openapi-typescript/blob/main/packages/openapi-typescript/LICENSE") and free for use.
+The code is [MIT-licensed](https://github.com/openapi-ts/openapi-typescript/blob/main/packages/openapi-typescript/LICENSE) and free for use.
```
Comment (yoshi2no) [inline docs/introduction.md:10]: This is the fix for the invalid link mentioned in the changes.

**Comment by drwpow** [comment]
Oh shoot it’s not the developer permissions; it’s that Cloudflare just doesn’t deploy previews for forks at all 🤔. My commit didn’t do anything. I’ve just always assumed.

Anyway, made a preview-only PR if you wanted to inspect it: https://github.com/openapi-ts/openapi-typescript/pull/1857

Overall, the dev version is pretty reliable! But just in case others wanted to inspect it easily too.

**Review by drwpow — APPROVED** [review]
This looks amazing! Great work with this, thank you! 🙏 Skimming through the pages, they seem good to me (minus me not being able to read Japanese). Docs all seem to be working well.

But let me know when you’re ready to merge this! This looks good from my end. After merge, I’ll trigger a manual reindex from Algolia so that appears in search as soon as possible.

**Comment by yoshi2no** [comment]
@drwpow 

Thank you for the review! 
After taking another quick look, everything seems good to me as well. Let’s go ahead and merge it! 🚀

**Comment by siketyan** [comment]
As a native Japanese speaker, thank you and great work @yoshi2no! I took a look the diff and the preview quickly and could not find any issues for now. I would like to review Japanese docs if needed, so feel free to mention me if you are finding a reviewer for that!

**Comment by yoshi2no** [comment]
@siketyan 

Thank you for your comment and for taking the time to review! 🙏 
I’ll definitely mention you if I need a reviewer in the future!

**Pull request merged**